### PR TITLE
Ensure custom tooltips are positioned correctly

### LINKED ISSUE #156: Unintentional breaking change in custom tooltip API
When writing #143, I forgot that orama allows consumers to pass a custom tooltip component, so the changes I made to the API of the default tooltip component should not have been backwards-incompatible. This caused an unintentionally breaking change in 2.0.1.

We can keep most of that refactor, but we need to revert the component API, which will require the measuring code to be pulled back out into another component.

### PR DESCRIPTION
**Pull request by nabeel-**
Closes #156

### CODE REVIEW

File: src/tooltips/TooltipContainer.js [hunk 1]
```diff
@@ -61,6 +62,7 @@ export default class TooltipContainer extends React.Component {
   renderTooltip() {
     const {hoverData, layerProps, theme} = this.props
     const {Tooltip = DefaultTooltip} = layerProps
+    const PositionedTooltip = withMeasuredSize(Tooltip, this.handleMeasure)
```
Comment (billyjanitsch) [inline src/tooltips/TooltipContainer.js]: This creates a new component in every render, which means that the tooltip needs to unmount and re-mount every render. Can we write the measuring as a component instead?

File: src/tooltips/Tooltip.js [hunk 2]
```diff
@@ -14,27 +14,9 @@ export default class Tooltip extends React.PureComponent {
   static propTypes = {
     hoverData: PropTypes.oneOfType([PropTypes.array, PropTypes.object]),
     layerProps: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
-    onMeasure: PropTypes.func.isRequired,
     theme: CustomPropTypes.theme.isRequired,
   }
```
Comment (billyjanitsch) [inline src/tooltips/Tooltip.js]: Can you compare this prop API against the one that existed for a custom tooltip before the refactor, and verify that it’s the same? This prop may not have been the only divergence

Comment (nabeel-) [reply src/tooltips/Tooltip.js]: Only other change was the `isRequired` on `theme`

File: src/tooltips/TooltipContainer.js (hunk 1 above)
Comment (nabeel-) [reply src/tooltips/TooltipContainer.js]: ✍️

**Pull request merged**